packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -769,6 +770,48 @@ export class CdkToolkit {
     // If there is an '--app' argument and an environment looks like a glob, we
     // select the environments from the app. Otherwise, use what the user said.
 
+    const environments = await this.defineEnvironments(userEnvironmentSpecs);
+
+    const limit = pLimit(20);
+
+    // eslint-disable-next-line @cdklabs/promiseall-no-unbounded-parallelism
+    await Promise.all(environments.map((environment) => limit(async () => {
+      success(' ⏳  Bootstrapping environment %s...', chalk.blue(environment.name));
+      try {
+        const result = await bootstrapper.bootstrapEnvironment(environment, this.props.sdkProvider, options);
+        const message = result.noOp
+          ? ' ✅  Environment %s bootstrapped (no changes).'
+          : ' ✅  Environment %s bootstrapped.';
+        success(message, chalk.blue(environment.name));
+      } catch (e) {
+        error(' ❌  Environment %s failed bootstrapping: %s', chalk.blue(environment.name), e);
+        throw e;
+      }
+    })));
+  }
+
+  /**
+   * Garbage collects assets from a CDK app's environment
+   * @param options Options for Garbage Collection
+   */
+  public async garbageCollect(userEnvironmentSpecs: string[], options: GarbageCollectionOptions) {
+    const environments = await this.defineEnvironments(userEnvironmentSpecs);
+
+    for (const environment of environments) {
```

Comment:
intentionally not parallelizing here to avoid overcomplicating but not sure if its the right decision

Reply:
Can always just throw a TODO "parallelize this" or something